File: backend-python/app/modules/system/dto/role_request.py
from pydantic import BaseModel, Field


class RoleSaveRequest(BaseModel):
    role_name: str = Field(alias="roleName", min_length=1, max_length=64)
    role_code: str = Field(alias="roleCode", min_length=1, max_length=64)
    status: int | None = Field(default=None, ge=0, le=1)
    remark: str | None = Field(default=None, max_length=255)


class MenuAssignRequest(BaseModel):
    menu_ids: list[int] = Field(alias="menuIds")

File: backend-rust/src/modules/system/dto/role_request.rs
use serde::Deserialize;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RoleSaveRequest {
    pub role_name: String,
    pub role_code: String,
    pub status: Option<i32>,
    pub remark: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RolePermissionAssignRequest {
    pub menu_ids: Vec<i64>,
}
